finance_manager/src/main/java/com/manager/finance/user/service/ProfileServiceImpl.java

```diff
@@ -16,33 +17,36 @@
 @RequiredArgsConstructor
 public class ProfileServiceImpl implements ProfileService {
     private final ProfileDao profileDao;
+    private final UserService userService;
     private final ProfileMapper profileMapper;
 
     @Override
-    public void deleteProfile(User id) {
-        Profile activeProfile = findByUserIdWithValidation(id);
+    public void deleteProfile() {
+        Profile activeProfile = findByUserIdWithValidation();
         activeProfile.setIsDeleted(true);
         profileDao.save(activeProfile);
```

Comment:
just for info, an alternative solution would be to use annotation on entity @SQLDelete